torch/_dynamo/variables/functions.py

```diff
@@ -510,7 +510,6 @@ def bind_args(self, parent, args, kwargs):
 
         for idx, name in enumerate(code.co_freevars):
             cell = self.closure.items[idx]
-            assert getattr(cell, name, name) == name
```

Comment:
This assert does not make sense at all. `cell` is a VariableTracker. The test failed because `name` here was `source` and variable tracker has a source.